charts/px-central/local_snapshot_restore.py

```diff
@@ -0,0 +1,499 @@
+            logger.error(f"Failed to scale down {volume_name} to HA=1: {str(e)}")
+            return
+    else:
+        logger.debug(f"Volume {volume_name} is already at HA={original_ha_value}, no scale-down needed.")
+
+    # Restore from snapshot
+    try:
+        restore_volume(volume_name, snapshot_id)
+    except VolumeRestoreError as e:
+        logger.error(f"Restore failed for {volume_name}: {str(e)}")
+        return
+
+    # Scale back up to the original HA if we scaled down
+    if changed_ha:
+        logger.debug(f"Reverting HA for {volume_name} back to {original_ha_value}")
+        current_ha = get_volume_ha(volume_name)
+        if current_ha is not None and current_ha != original_ha_value:
+            try:
+                reduce_volume_ha_stepwise_excluding_snapshot(
+                    volume_name,
+                    current_ha,
+                    original_ha_value,
+                    snapshot_id
+                )
+            except VolumeUpdateError as e:
+                logger.error(f"Failed to revert HA for {volume_name} back to {original_ha_value}: {str(e)}")
+
+def main(input_file):
+    """
+    1) Read volumes from the JSON file.
+    2) Dynamically compute widths for 'Index', 'Namespace', 'PVC', 'Name', 'SnapshotID'.
+    3) Print a table with a top border, header, data rows, and bottom border.
+    4) Ask user which volumes to restore (by index or 'all').
+    5) For each selected volume, do the restore logic.
+    """
+    # ---------------------------------------------------------------------
+    # Read JSON
+    # ---------------------------------------------------------------------
+    try:
+        with open(input_file, 'r') as f:
+            data = json.load(f)
+    except Exception as e:
+        logger.error(f"Failed to read or parse {input_file}: {e}")
+        return 1  # Return a non-zero code to indicate failure
+
+    if not isinstance(data, dict):
+        logger.error("Top-level JSON is not a dictionary.")
+        return 1
+
+    backup_info = data.get("backup_info", {})
+    if not isinstance(backup_info, dict):
+        logger.error("`backup_info` is not a dictionary.")
+        return 1
+
+    volumes = backup_info.get("volumes", [])
+    if not isinstance(volumes, list):
+        logger.error("`volumes` is not a list.")
+        return 1
+
+    if not volumes:
+        logger.error("No volumes found in the input file.")
+        return 1
+
+    # ---------------------------------------------------------------------
+    # Build table data
+    # ---------------------------------------------------------------------
+    table_data = []
+    for i, vol in enumerate(volumes):
+        if not isinstance(vol, dict):
+            logger.warning(f"Volume at index {i} is not a dictionary. Skipping...")
+            continue
+
+        vol_namespace = vol.get("namespace", "")
+        vol_pvc = vol.get("pvc", "")
+        vol_name = vol.get("name", "")
```

Comment:
Should we add a check if any of the driver type is not pxd, to bail put from here?

Reply:
Yes, I am waiting for the @sgajawada-px PR where he is adding a flag in the backup object. I am planning to check for that flag and if it is not set, will return upfront.